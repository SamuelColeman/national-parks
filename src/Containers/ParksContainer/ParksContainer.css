.parks_container {
	display: grid;
  background: #003300;
  height: 100vh;
  text-align: center;
  width: 100%;
  grid-template-areas: 
		"header search search" 
		"header article article"
		"header article article";
	grid-template-columns: 20% auto;
	grid-template-rows: 10% auto;
}

.parks_container-header {
	grid-area: header;
	font-size: 5vh;
  color: white;
  background-color: #331a00;
  height: 100vh;
  margin: 0px;
}

.parks_container-form {
	grid-area: search;
}

.parks_container-cards {
	grid-area: article;
	max-height: 100vh;
	overflow: scroll;
	display: flex;
  flex-wrap: wrap;
  justify-content: center;
}