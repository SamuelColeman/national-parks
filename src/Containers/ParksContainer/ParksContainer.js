import React from 'react';
import ParkCard from '../ParkCard/ParkCard';
import './ParksContainer.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const ParksContainer = ({ parks, selectedState, displayParkInfo, handleSearch, errorMsg, loading, submitState }) => {
	const mappedParks = parks.map(park => {
		return <ParkCard {...park} page={false} displayParkInfo={displayParkInfo}/>
	})
	return (
		<section className='parks_container'>
			<section className='parks_container-header'>
				<h1 className='parks_container-header-title'>National Parks: {selectedState}</h1>
				<Link to='/'>
					<button className='parks_container-header-button'>Select State</button>
				</Link>
			</section>
			<form className='parks_container-form'>
			<button className='parks_container-search-btn' type='button' onClick={() => submitState(selectedState)}>All</button>
				<input className='parks_container-search' onChange={(e) => handleSearch(e)} type='text' placeholder='Search'/>
			</form>
			<section className='parks_container-cards'>
				{loading && <h1 className='parks_container-loading'>Loading...</h1>}
				{errorMsg && <h1>{errorMsg}</h1>}
				<div className='parks_container-cards'>{mappedParks}</div>
			</section>
		</section>
	)
}

export const mapStateToProps = (state) => ({
  parks: state.parks,
  selectedState: state.selectedState,
  errorMsg: state.errorMsg,
  loading: state.loading
})

export default connect(mapStateToProps)(ParksContainer);