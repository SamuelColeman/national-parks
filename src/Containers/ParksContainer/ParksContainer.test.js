import React from 'react';
import { shallow } from 'enzyme';
import { ParksContainer, mapStateToProps } from './ParksContainer';

describe('ParksContainer', () => {
	
	let wrapper;
	let mockParks = [{ 
    	description: "Historic Site",
			designation: "National Historic Site",
			directionsInfo: "Drive Car",
			directionsUrl: "http://www.directions.com",
			fullName: "National Historic Site",
			id: "2",
			latLong: "lat:36, long:82",
			name: "Historic Park",
			parkCode: "pant",
			states: "ZN",
			url: "https://www.example.com",
			weatherInfo: "Sunny"
		}];
		let mockSelectedState = 'ZN';
		let mockParkId = '5D';
		let mockDisplayParkInfo = jest.fn();
		let mockHandleSearch = jest.fn();
		let mockErrorMsg = 'Invalid State';
		let mockLoading = false;
		let mockSubmitState = jest.fn();

	beforeEach(() => {
    wrapper = shallow(<ParksContainer 
    	parks={mockParks}
    	selectedState={mockSelectedState}
    	displayParkInfo={mockDisplayParkInfo}
    	handleSearch={mockHandleSearch}
    	errorMsg={mockErrorMsg}
    	loading={mockLoading}
    	submitState={mockSubmitState}
    	/>)
 	})

  it('should match snapshot with correct data passing through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with correct data passing through', () => {
  	mockLoading = true;
    expect(wrapper).toMatchSnapshot();
  });

  it('should call submitState when all button is clicked', () => {
    wrapper.find('button').at(1).simulate('click');

    expect(mockSubmitState).toHaveBeenCalledWith(mockSelectedState);
    });

  it('should call handleSearch when search input field is changed', () => {
  	const mockEvent = {target: {value: 'TN'}};

    wrapper.find('input').simulate('change', mockEvent);

    expect(mockHandleSearch).toHaveBeenCalledWith(mockEvent);
    });

  it('map state to props gives the parks array in state', () => {
		const mockState = { parks: mockParks};

		const expected = {
			parks: mockState.parks
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the selected state in state', () => {
		const mockState = { selectedState: mockSelectedState};

		const expected = {
			selectedState: mockState.selectedState
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the error message in state', () => {
		const mockState = { errorMsg: mockErrorMsg};

		const expected = {
			errorMsg: mockState.errorMsg
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the loading in state', () => {
		const mockState = { loading: mockLoading};

		const expected = {
			loading: mockState.loading
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});
});