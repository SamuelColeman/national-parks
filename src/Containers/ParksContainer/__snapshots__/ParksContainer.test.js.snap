// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParksContainer should match snapshot with correct data passing through 1`] = `
<section
  className="parks_container"
>
  <section
    className="parks_container-header"
  >
    <h1
      className="parks_container-header-title"
    >
      National Parks: 
      ZN
    </h1>
    <Link
      to="/"
    >
      <button
        className="parks_container-header-button"
      >
        Select State
      </button>
    </Link>
  </section>
  <form
    className="parks_container-form"
  >
    <button
      className="parks_container-search-btn"
      onClick={[Function]}
      type="button"
    >
      All
    </button>
    <input
      className="parks_container-search"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </form>
  <section
    className="parks_container-cards"
  >
    <h1>
      Invalid State
    </h1>
    <div
      className="parks_container-cards"
    >
      <Connect(ParkCard)
        description="Historic Site"
        designation="National Historic Site"
        directionsInfo="Drive Car"
        directionsUrl="http://www.directions.com"
        displayParkInfo={[MockFunction]}
        fullName="National Historic Site"
        id="2"
        latLong="lat:36, long:82"
        name="Historic Park"
        page={false}
        parkCode="pant"
        states="ZN"
        url="https://www.example.com"
        weatherInfo="Sunny"
      />
    </div>
  </section>
</section>
`;

exports[`ParksContainer should match snapshot with correct data passing through 2`] = `
<section
  className="parks_container"
>
  <section
    className="parks_container-header"
  >
    <h1
      className="parks_container-header-title"
    >
      National Parks: 
      ZN
    </h1>
    <Link
      to="/"
    >
      <button
        className="parks_container-header-button"
      >
        Select State
      </button>
    </Link>
  </section>
  <form
    className="parks_container-form"
  >
    <button
      className="parks_container-search-btn"
      onClick={[Function]}
      type="button"
    >
      All
    </button>
    <input
      className="parks_container-search"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </form>
  <section
    className="parks_container-cards"
  >
    <h1>
      Invalid State
    </h1>
    <div
      className="parks_container-cards"
    >
      <Connect(ParkCard)
        description="Historic Site"
        designation="National Historic Site"
        directionsInfo="Drive Car"
        directionsUrl="http://www.directions.com"
        displayParkInfo={[MockFunction]}
        fullName="National Historic Site"
        id="2"
        latLong="lat:36, long:82"
        name="Historic Park"
        page={false}
        parkCode="pant"
        states="ZN"
        url="https://www.example.com"
        weatherInfo="Sunny"
      />
    </div>
  </section>
</section>
`;
